assimilation_code/modules/io/direct_netcdf_mod.f90

```diff
@@ -1619,7 +1637,7 @@ function create_and_open_state_output(name_handle, dom_id, copy_number, &
 call nc_check(ret, routine, 'nf90_create "'//trim(filename)//'"')
 
 ret = nf90_enddef(ncfile_out)
-call nc_check(ret, routine, 'end define mode')
+call nc_check(ret, routine, 'end define mode before writing grid info')
```

Comment:
some models don't have a 'grid' per-se.   nc_write_model_atts() calls model_mod code and it can write any number of variables into the netcdf file, including elevation, masks, etc - any non-state fields that it needs.   i don't think this error message change is accurate.  i'd also update the comment on line 1642 to say something like 'write additional model-specific information' because it's typically more than grid info.